File: compiled_lang/rust/h10.associative.rs
use std::collections::HashMap;

fn main() {
    // initialize map with key/value pairs
    let mut ages: HashMap<&str, i32> = HashMap::from([
        ("bob", 34), ("ed", 58), ("steve", 32), ("ralph", 23),
    ]);
    // append another set of key/value pairs into map
    let more: HashMap<&str, i32> = HashMap::from([("deb", 46), ("kate", 19)]);
    ages.extend(more);

    // iterate through map by keys, print key/value pairs
    println!("The ages are: ");
    for (name, age) in &ages {
        println!(" ages[{}]={}", name, age);
    }
}

File: compiled_lang/cs/src/h10.associative.cs
using System;
using System.Collections.Generic;

class H10Associative
{
    static void Main()
    {
        // initialize dictionary with key/value pairs
        var ages = new Dictionary<string, int> {
            { "bob", 34 }, { "ed", 58 }, { "steve", 32 }, { "ralph", 23 }
        };
        // append another set of key/value pairs into dictionary
        var more = new Dictionary<string, int> { { "deb", 46 }, { "kate", 19 } };
        foreach (var pair in more) ages[pair.Key] = pair.Value;

        // iterate through dictionary by keys, print key/value pairs
        Console.WriteLine("The ages are: ");
        foreach (var name in ages.Keys)
        {
            Console.WriteLine($" ages[{name}]={ages[name]}");
        }
    }
}
